dbt_semantic_interfaces/dbt_semantic_interfaces/objects/constraints/filter_renderer.py

```diff
@@ -0,0 +1,203 @@
+    that can be used in a SQL query like "user__home_state_latest IN ('CA', 'HI', 'WA')"
+
+    This class does not use the *spec classes as they are only available in the metricflow package.
+    """
+
+    # Names of the function calls used in the Jinja template.
+    _DIMENSION_FUNCTION_NAME = "dimension"
+    _TIME_DIMENSION_FUNCTION_NAME = "time_dimension"
+    _IDENTIFIER_FUNCTION_NAME = "identifier"
+
+    # To extract the parameters to the calls, we use a function to record the parameters while rendering the Jinja
+    # template. The rendered result is not used, but since Jinja has to render something, using this as a placeholder.
+    _DUMMY_PLACEHOLDER = "DUMMY_PLACEHOLDER"
+
+    @staticmethod
+    def extract_parameter_sets(templated_filter: str) -> FilterCallParameterSets:
+        """Parse the filter and extract the metric object call parameters.
+
+        An abbreviated example:
+
+        "{{ dimension('home_state_latest', entity_path=['user']) }} IN ('CA', 'HI', 'WA')"
+
+        ->
+
+        FilterCallParameterSets(
+            dimension_call_parameter_sets=(
+                DimensionCallParameterSet(
+                    entity_path=("user",),
+                    dimension_reference="home_state_latest",
+                ),
+            )
+            ...
+        )
+
+        """
+
+        dimension_call_parameter_sets: List[DimensionCallParameterSet] = []
+        time_dimension_call_parameter_sets: List[TimeDimensionCallParameterSet] = []
+        entity_call_parameter_sets: List[EntityCallParameterSet] = []
+
+        class _DummyCallRenderer(FilterFunctionCallRenderer):
+            def render_dimension_call(self, dimension_call_parameter_set: DimensionCallParameterSet) -> str:  # noqa: D
+                dimension_call_parameter_sets.append(dimension_call_parameter_set)
+                return FilterRenderer._DUMMY_PLACEHOLDER
+
+            def render_time_dimension_call(  # noqa: D
+                self, time_dimension_call_parameter_set: TimeDimensionCallParameterSet
+            ) -> str:
+                time_dimension_call_parameter_sets.append(time_dimension_call_parameter_set)
+                return FilterRenderer._DUMMY_PLACEHOLDER
+
+            def render_entity_call(self, entity_call_parameter_set: EntityCallParameterSet) -> str:  # noqa: D
+                entity_call_parameter_sets.append(entity_call_parameter_set)
+                return FilterRenderer._DUMMY_PLACEHOLDER
+
+        FilterRenderer.render(
+            templated_filter_sql=templated_filter,
+            call_renderer=_DummyCallRenderer(),
+        )
+
+        return FilterCallParameterSets(
+            dimension_call_parameter_sets=tuple(dimension_call_parameter_sets),
+            time_dimension_call_parameter_sets=tuple(time_dimension_call_parameter_sets),
+            entity_call_parameter_sets=tuple(entity_call_parameter_sets),
+        )
```

Comment:
This is super weird. Maybe this is the only reasonable way to do this? Either way, I guess it'll do for now but it's really odd to overload the renderer for this extraction.

Reply:
There is a way to do this by parsing the AST that Jinja can generate for you. However, it seemed like a more complex approach so I didn't go into it too deeply. I agree that this is a little odd though.